File: app/src/main/java/pl/futuredev/capstoneproject/database/dao/CityDao.java
package pl.futuredev.capstoneproject.database.dao;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.List;

import pl.futuredev.capstoneproject.database.entity.CityPOJO;

@Dao
public interface CityDao {

    @Query("SELECT * FROM city")
    LiveData<List<CityPOJO>> loadAllCities();

    @Query("SELECT * FROM city")
    List<CityPOJO> loadAllCitiesSync();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertCity(CityPOJO cityPOJO);

    @Delete
    void deleteCity(CityPOJO cityPOJO);

    @Query("SELECT * FROM city WHERE name = :name")
    LiveData<CityPOJO> loadCityById(String name);

    @Query("SELECT * FROM city WHERE name = :name")
    CityPOJO loadCityByNameCityPOJO(String name);

}

File: app/src/main/java/pl/futuredev/capstoneproject/data/local/CityDao.kt
package pl.futuredev.capstoneproject.data.local

import androidx.lifecycle.LiveData
import androidx.room.*
import pl.futuredev.capstoneproject.data.local.entities.City
import pl.futuredev.capstoneproject.others.Resource

@Dao
interface CityDao {

    @Query("SELECT * FROM city")
    suspend fun getAllCities(): List<City>

    @Query("SELECT * FROM city WHERE name =:name LIMIT 1")
    fun getCityByName(name:String): LiveData<City>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertCity(city: City)

    @Query("DELETE FROM city WHERE name =:name")
    suspend fun deleteCityByName(name: String)

}